File: ios/Wellvo/Models/WellvoAlert.swift
import Foundation

struct WellvoAlert: Codable, Identifiable {
    let id: UUID
    let familyId: UUID
    let receiverId: UUID
    let type: String
    let title: String
    let message: String
    let data: [String: Double]?
    var isRead: Bool
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, type, title, message, data
        case familyId = "family_id"
        case receiverId = "receiver_id"
        case isRead = "is_read"
        case createdAt = "created_at"
    }
}

File: android/app/src/main/java/net/wellvo/android/data/models/WellvoAlert.kt
package net.wellvo.android.data.models

import androidx.compose.runtime.Immutable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Serializable
data class WellvoAlert(
    val id: String,
    @SerialName("family_id")
    val familyId: String,
    @SerialName("receiver_id")
    val receiverId: String,
    val type: String,
    val title: String,
    val message: String,
    val data: Map<String, Double>? = null,
    @SerialName("is_read")
    val isRead: Boolean,
    @SerialName("created_at")
    val createdAt: String
)
